src/grblogtools/api.py

```diff
@@ -75,18 +75,28 @@ def summary(self):
         return summary
 
     def parse(self, logfile: str) -> None:
-        """Parse a single log file, containing the log from one run.
-
-        TODO extend this to also check multiple logs from one file.
-        """
+        """Parse a single file. The log file may contain multiple run logs."""
         parser = SingleLogParser()
+        subsequent = SingleLogParser()
+        log_count = 0
         with open(logfile) as infile:
             lines = iter(infile)
             for line in lines:
-                if parser.start_parsing(line):
-                    for line in lines:
-                        parser.continue_parsing(line)
-        self.parsers.append((logfile, parser))
+                if not parser.parse(line):
+                    assert not subsequent.started
+                    if subsequent.parse(line):
+                        # The current parser did not match but an empty parser
+                        # matched a header line.
+                        log_count += 1
+                        self.parsers.append((f"{logfile}({log_count})", parser))
```

Comment:
Is there any specific reason for starting the naming from 1 and not 0? If we adapt the naming convention `{log file}{log_count}` even for the case where there is only one log, we can simplify the logic at the end and there is no need for an `if` statement at the end.

Reply:
No particular reason, I'm following the merged_logs=True behaviour of the current code if there are multiple logs.

But now that you mention it, perhaps it would be cleaner to keep the LogFilePath and the index as separate columns instead of munging them together? That way it still accurately reflects the source file. We could keep LogIndex (where *n* indicates the nth log in the file) as its own column?

Reply:
I agree. This is a good suggestion.

Reply:
Ok, changed in a279bfe and merged.